models/pose_estimation_mediapipe/demo.cpp

```diff
@@ -0,0 +1,2913 @@
+        make_pair(0, 4), make_pair(4, 5), make_pair(5, 6), make_pair(6, 8),
+        make_pair(9, 10),
+        make_pair(12, 14), make_pair(14, 16), make_pair(16, 22), make_pair(16, 18), make_pair(16, 20), make_pair(18, 20),
+        make_pair(11, 13), make_pair(13, 15), make_pair(15, 21), make_pair(15, 19), make_pair(15, 17), make_pair(17, 19),
+        make_pair(11, 12), make_pair(11, 23), make_pair(23, 24), make_pair(24, 12),
+        make_pair(24, 26), make_pair(26, 28), make_pair(28, 30), make_pair(28, 32), make_pair(30, 32),
+        make_pair(23, 25), make_pair(25, 27),make_pair(27, 31), make_pair(27, 29), make_pair(29, 31) };
+    for (auto p : segment)
+        if (keeplandmarks.at<uchar>(p.first) && keeplandmarks.at<uchar>(p.second))
+            line(image, Point(landmarks.row(p.first)), Point(landmarks.row(p.second)), Scalar(255, 255, 255), thickness);
+    if (isDrawPoint)
+        for (int idxRow = 0; idxRow < landmarks.rows; idxRow++)
+            if (keeplandmarks.at<uchar>(idxRow))
+                circle(image, Point(landmarks.row(idxRow)), thickness, Scalar(0, 0, 255), -1);
+}
+
+
+pair<Mat, Mat> visualize(Mat image, vector<tuple<Mat, Mat, Mat, Mat, Mat, float>> poses)
+{
+    Mat displayScreen = image.clone();
+    Mat display3d(400, 400, CV_8UC3, Scalar::all(0));
+    line(display3d, Point(200, 0), Point(200, 400), Scalar(255, 255, 255), 2);
+    line(display3d, Point(0, 200), Point(400, 200), Scalar(255, 255, 255), 2);
+    putText(display3d, "Main View", Point(0, 12), FONT_HERSHEY_DUPLEX, 0.5, Scalar(0, 0, 255));
+    putText(display3d, "Top View", Point(200, 12), FONT_HERSHEY_DUPLEX, 0.5, Scalar(0, 0, 255));
+    putText(display3d, "Left View", Point(0, 212), FONT_HERSHEY_DUPLEX, 0.5, Scalar(0, 0, 255));
+    putText(display3d, "Right View", Point(200, 212), FONT_HERSHEY_DUPLEX, 0.5, Scalar(0, 0, 255));
+    bool isDraw = false;  // ensure only one person is drawn
+
+    for (auto pose : poses)
+    {
+        Mat bbox = get<0>(pose);
+        Mat landmarksScreen = get<1>(pose);
+        Mat landmarksWord = get<2>(pose);
+        Mat mask;
+        get<3>(pose).convertTo(mask, CV_8U);
+        Mat heatmap = get<4>(pose);
+        float conf = get<5>(pose);
+        Mat edges;
+        Canny(mask, edges, 100, 200);
+        Mat kernel(2, 2, CV_8UC1, Scalar::all(1)); // expansion edge to 2 pixels
+        dilate(edges, edges, kernel);
+        Mat edgesBGR;
+        cvtColor(edges, edgesBGR, COLOR_GRAY2BGR);
+        Mat idxSelec = edges == 255;
+        edgesBGR.setTo(Scalar(0, 255, 0), idxSelec);
+
+        add(edgesBGR, displayScreen, displayScreen);
+        // draw box
+        Mat box;
+        bbox.convertTo(box, CV_32S);
+
+        rectangle(displayScreen, Point(box.row(0)), Point(box.row(1)), Scalar(0, 255, 0), 2);
+        putText(displayScreen, format("%4f", conf), Point(box.at<int>(0, 0), box.at<int>(1, 0) + 12), FONT_HERSHEY_DUPLEX, 0.5, Scalar(0, 0, 255));
```

Comment:
The point should be `Point(box.at<int>(0, 0), box.at<int>(0, 1) + 12)`

Reply:
I wrote values at top of the screen like this values are not moving as box changed

Reply:
Some models support muti-person detection so the score will not only one. BUT, the person detection model from MediaPipe only support one person detection, the performance of muti-person detection is bad. So I think it's OK.